lib/jekyll/assets/cached.rb

```diff
@@ -15,8 +15,13 @@ class Cached < Sprockets::CachedEnvironment
       def initialize(env)
         @parent = env
         @jekyll = env.jekyll
+        @resolve_cache = {}
         super env
       end
+
+      def resolve(*args)
+        @resolve_cache[args] ||= super
```

Comment:
And on that note, that is actually part of the problem that was mentioned upstream that they couldn't reliably solve because Sprockets 3 was so complicated.  In Sprockets 4 they resolved this by expanding every path and caching the absolute path 100% of the time, they tried that in Sprockets 3 but couldn't get it done so it's stuck with what you experienced.